src/styles/main.scss

```diff
@@ -0,0 +1,66 @@
+body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+$day-size: 100px;
+$gap: 1px;
+
+.calendar {
+  max-width: calc($day-size * 7 + $gap * 6);
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 10px;
+  &__day {
+    background-color: #eee;
+    position: relative;
+    box-sizing: border-box;
+    width: $day-size;
+    height: $day-size;
+    border: 1px solid black;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    &:hover {
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+      cursor: pointer;
+      transition-duration: 0.5s;
+    }
+  }
+}
+
+@for $i from 1 to 32 {
+  .calendar :nth-child(#{$i})::before {
```

Comment:
The selector `.calendar :nth-child(#{$i})::before` might be too broad, as it applies to any child element of `.calendar`. Ensure this is the intended behavior, or consider narrowing the selector if needed.